docs/src/architecture/transactions.md

```diff
@@ -1,45 +1,49 @@
 # Transactions
-Asset transfers between accounts are done by executing transactions. Miden aims for parallel and private transaction execution. Therefore, a transaction is always performed against a single account which provides asynchronicity. And, every transaction causes a provable state-change and a STARK proof thereof. Thus, when executed locally, it provides privacy.
+Transactions in Miden can be understood as facilitating account state changes. Asset transfers between accounts are done by executing transactions. Basically, they take one single account and some notes as input and output the same account at a new state together with some other notes. Miden aims for parallel and private transaction execution. Because a transaction is always performed against a single account Miden obtains asynchronicity. And, because every transaction causes a provable state-change and a STARK proof thereof it provides privacy, when executed locally.
```

Comment:
The last two sentences here are a bit confusing:

- I'm not sure people will understand what "asynchronicity" is or whether it is good or bad. I'd either remove this sentence or rewrite it to make it easier to understand the implications.
- The contents in the last sentence are fine but it is structured in a complex way (e.g., I would not use words like "thereof").

Reply:
Indeed I changed it. And let's improve on this later.